src/emulator/dataconnect/pgliteServer.ts

```diff
@@ -19,11 +21,16 @@ import { logger } from "../../logger";
 export class PostgresServer {
   private username: string;
   private database: string;
+  private dataDirectory?: string;
+  private importPath?: string;
 
-  public db: PGlite | undefined;
+  public db: PGlite | undefined = undefined;
   public async createPGServer(host: string = "127.0.0.1", port: number): Promise<net.Server> {
-    const db: PGlite = await this.getDb();
-    await db.waitReady;
+    const getDb: () => Promise<PGlite> = () => {
```

Comment:
nit: How about `this.getDb.bind(this)`?

Reply:
Ahh, that was what I was looking for, thanks! Despite many years of JS, `this` still screws me up 😬